onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.h

```diff
@@ -20,19 +20,15 @@ struct NodeGroup {
   NodeIndex target_node;
 };
 
-// Base QDQ checker. Finds and provides the DQ and Q nodes to the operator specific checkers, as the QDQ optimizations
-// always involve those nodes.
-class BaseSelector : public NodeSelector {
+class NodeGroupSelector {
```

Comment:
Overall LGTM, only question is that if we have BaseSelector for SAT and NodeGroupSelector for EP 's QDQSelectors, this may make sharing selector construction a bit difficult

Reply:
Yeah - I don't think there's a good way to share the construction. Sharing all the implementation cleanly is more important (bulk of the binary size cost is there) and the selector construction may diverge anyway based on which selectors are applicable to an EP.